qiskit/transpiler/passes/scheduling/padding/pad_delay.py

```diff
@@ -83,8 +86,17 @@ def _pad(
         next_node: DAGNode,
         prev_node: DAGNode,
     ):
-        if not self.fill_very_end and isinstance(next_node, DAGOutNode):
-            return
-
-        time_interval = t_end - t_start
-        self._apply_scheduled_op(dag, t_start, Delay(time_interval, dag._unit), qubit)
+        # NOTE: The :meth:`._pad` method is currently being ported as it is the core logic within the ``PadDelay`` pass
+        # and the :meth:`.run` method would need to be ported along with the :class:`.BasePadding` class when the ``PadDynamicDecoupling`` pass
+        # is finished being ported.
+
+        pad_delay(
+            dag,
+            qubit,
+            t_start,
+            t_end,
+            self.fill_very_end,
+            next_node,
+            prev_node,
+            self.property_set,
```

Comment:
and finally, you may pass the unit here, as follows:

```suggestion
            self.property_set,
            dag._unit,
```


This overall should fix the problem of reusing a deprecated library, which is not preferable at all.